internal/pkg/service/cli/cmd/remote/create.go

```diff
@@ -60,7 +59,7 @@ func CreateBranchCommand(p dependencies.Provider) *cobra.Command {
 			}
 
 			// Send cmd successful/failed event
-			defer func() { d.EventSender().SendCmdEvent(d.CommandCtx(), start, cmdErr, "remote-create-branch") }()
+			defer d.EventSender().SendCmdEvent(d.CommandCtx(), time.Now(), &cmdErr, "remote-create-branch")
```

Comment:
Tu bola chyba, mozem to asi podrobne vysvetlit na standupe.

-------------

1. V `SendCmdEvent` je `recover()` co ma zachytit `panic`:
- Ale vobec to nefungovalo.
- `recover` sa moze totiz volat iba priamo vo funkcii, ktora je zadana v `defer,` v tomto pripade ta anonymna `func() {`
- ide: https://go.dev/play/p/YKxuRlF0S1G
- nejde: https://go.dev/play/p/zInJ1j3v7Yb
- see: https://stackoverflow.com/questions/49344478/why-recover-does-not-work-in-a-nested-deferred-function

Bud som to vtedy nevedel, alebo mi to nedoslo.

-------------------

2. Ako som to upravil, tak namiesto `error`, musim pouzit `*error`.
- Pretoze dane parametre sa resolvuju v case definicie `defer`, nie az pri vykonani.
- A vtedy je `err` este vzdy `nil`, nastavi sa az neskor.
- Preto pouzijem pointer.


To iste mame v telemetrii (vid `defer EndSpan`):
https://github.com/keboola/keboola-as-code/blob/b0d067a49bab567c161d936af85f92f4759dda2e/pkg/lib/operation/repository/pull/operation.go#L23-L25

https://github.com/keboola/keboola-as-code/blob/b0d067a49bab567c161d936af85f92f4759dda2e/internal/pkg/telemetry/span.go#L22-L33

----------

Tu je este dobre spomenut, ze:
- `var errPtr *error = nil` nie je to iste ako 
- `var err error = nil`
- preto sa v tej `telemetry` ^^^ kontroluje `nil` 2x, jedno je `nil` pointer, a druhe je `nil` interface, ... tj. iny typ `nil`.

Reply:
BTW: meranie casu je teraz presnejsie, pretoze `(start =) time.Now()` sa vola az po ukonceni dialogov, cas teda viac  zodpoveda samotnej operacii.